src/main/kotlin/dev/encore/intellij/StartupActivity.kt

```diff
@@ -0,0 +1,46 @@
+package dev.encore.intellij
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.startup.StartupActivity
+import com.intellij.util.text.SemVer
+import dev.encore.intellij.sqldb.Detector
+import dev.encore.intellij.utils.isInEncoreApp
+import dev.encore.intellij.utils.runEncoreCommand
+
+/*
+ * Startup Activity allows us to run blocking external calls to the encore daemon to extract information
+ * about the project, which we can then store and reference at runtime
+ */
+class StartupActivity : StartupActivity.Background, StartupActivity.RequiredForSmartMode {
+    override fun runActivity(project: Project) {
+        // Skip our startup if it's not an Encore app
+        if (!isInEncoreApp(project)) {
+            return
+        }
+
+        Encore.LOG.info("Performing startup activity for Encore...")
+
+        // Verify Encore is installed
+        val versionOutput = runEncoreCommand(project, "version") ?: return
+        if (!versionOutput.startsWith("encore version ")) {
```

Comment:
Maybe better to just check for zero exit code? Or we should document this requirement in the CLI docs so we don't end up changing this without realizing the effects

Reply:
runEncoreCommand already checks the exit code and returns null if there's a bad exit; but I want to verify that the binary selected is;

1. Encore
2. containing a version number we can parse out for SemVer checks